DOCSP-30635 Mapping Rules migration (#24)
mongodb/docs-relational-migrator: source/mapping-rules/schema-mapping.txt

==============
Schema Mapping
==============

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Schema mapping is the process by which you decide how your source relational schema 

> should be represented in your target MongoDB database. You facilitate the schema 

Reviewer: (wording) 
Consider just "Facilitate the schema ..." 
Reasoning is to avoid redundancy.

Author: Done.

mapping process in Relational Migrator through customized mappings rules.

Mapping Rules
-------------

> During the schema mapping design process, Relational Migrator will automatically derive 

Reviewer: ```suggestion
During the schema mapping design process, Relational Migrator automatically derives 
```

Author: Done

the source data model from the schema of your relational database. The destination data 
model can be influenced by customizing:

- Mapping Rule options

> - Project Id field options

Reviewer: ```suggestion
- Project ID field options
```

.. image:: /img/schema-mapping/schema-mapping-relational-migrator-concept-diagram-b.png
   :figwidth: 450 px
   :alt: Schema Mapping Concept

Example
-------

> This provides a scenario and example implementation for the schema mapping process. 

Reviewer: (missing word?)
This [section]

Author: Done.

> This example will be converting a relational data model to a MongoDB data model.

Reviewer: ```suggestion
This example converts a relational data model to a MongoDB data model.
```

Using the ``MongoEnterprises`` relational database, the goal is to flatten 

> all of the databases tables so that all the order and customer data will be 

Reviewer: ```suggestion
all of the databases tables, so that all the order and customer data is 
```

Author: Done.

available in a single MongoDB collection.

> Use-Case Denormalizing Table Structure

Reviewer: (nested sections) 
I _think_ the rest of the sections should be nested.

Author: Done.

--------------------------------------

> The following image shows the relational data model which will be denormalized using the 

Reviewer: ```suggestion
The following image shows the relational data model to be denormalized using the 
```

Author: Done.

Relational Migrator.

.. image:: /img/schema-mapping/schema-mapping-mongo-enterprises.png
   :figwidth: 450 px
   :alt: Relational Data Model

Mapping Rule Results
--------------------

> 

Reviewer: (wording)
Consider a minor tweak: When using Relational Migrator, the goal is to ...

Author: Done.

The goal when using Relational Migrator is to achieve the following MongoDB data model. 
The ``Order`` collection contains all the child elements from the ``Customer``, 

> ``Order``, ``Product`` and ``Order Line`` tables. The collection will have nested 
> objects for the migrated tables. The results look like this:

Reviewer: ```suggestion
``Order``, ``Product`` and ``Order Line`` tables. The collection has nested 
objects for the migrated tables. The results look like this:
```

Author: Done
